rm-datasource/src/main/java/io/seata/rm/datasource/sql/druid/postgresql/PostgresqlInsertRecognizer.java

```diff
@@ -0,0 +1,122 @@
+import com.alibaba.druid.sql.ast.statement.SQLExprTableSource;
+import com.alibaba.druid.sql.ast.statement.SQLInsertStatement;
+import com.alibaba.druid.sql.dialect.postgresql.ast.stmt.PGInsertStatement;
+import com.alibaba.druid.sql.dialect.postgresql.visitor.PGOutputVisitor;
+import io.seata.rm.datasource.sql.SQLInsertRecognizer;
+import io.seata.rm.datasource.sql.SQLParsingException;
+import io.seata.rm.datasource.sql.SQLType;
+import io.seata.rm.datasource.sql.druid.BaseRecognizer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author japsercloud
+ */
+public class PostgresqlInsertRecognizer extends BaseRecognizer implements SQLInsertRecognizer {
+
+    private final PGInsertStatement ast;
+
+    /**
+     * Instantiates a new Postgresql insert recognizer.
+     *
+     * @param originalSQL the original sql
+     * @param ast the ast
+     */
+    public PostgresqlInsertRecognizer(String originalSQL, SQLStatement ast) {
+        super(originalSQL);
+        this.ast = (PGInsertStatement) ast;
+    }
+
+    @Override
+    public SQLType getSQLType() {
+        return SQLType.INSERT;
+    }
+
+    @Override
+    public String getTableAlias() {
+        return ast.getTableSource().getAlias();
+    }
+
+    @Override
+    public String getTableName() {
+        StringBuilder sb = new StringBuilder();
+        PGOutputVisitor visitor = new PGOutputVisitor(sb) {
+
+            @Override
+            public boolean visit(SQLExprTableSource x) {
+                printTableSourceExpr(x.getExpr());
+                return false;
+            }
+        };
+        visitor.visit(ast.getTableSource());
+        return sb.toString();
+    }
+
+    @Override
+    public List<String> getInsertColumns() {
+        List<SQLExpr> columnSQLExprs = ast.getColumns();
+        if (columnSQLExprs.size() == 0) {
+            // INSERT INTO ta VALUES (...), without fields clarified
+            return null;
+        }
+        List<String> list = new ArrayList<>(columnSQLExprs.size());
+        for (SQLExpr expr : columnSQLExprs) {
+            if (expr instanceof SQLIdentifierExpr) {
+                list.add(((SQLIdentifierExpr) expr).getName().toUpperCase());
+            } else {
+                throw new SQLParsingException("Unknown SQLExpr: " + expr.getClass() + " " + expr);
+            }
+        }
+        return list;
+    }
+
+    @Override
+    public List<List<Object>> getInsertRows() {
+        List<SQLInsertStatement.ValuesClause> valuesClauses = ast.getValuesList();
+        List<List<Object>> rows = new ArrayList<>(valuesClauses.size());
+        for (SQLInsertStatement.ValuesClause valuesClause : valuesClauses) {
+            List<SQLExpr> exprs = valuesClause.getValues();
+            List<Object> row = new ArrayList<>(exprs.size());
+            for (SQLExpr expr : valuesClause.getValues()) {
+                if (expr instanceof SQLValuableExpr) {
```

Comment:
please check out with MysqlInsertRecognizer and make sure it can suit to different situation in the insert.